### PR TITLE
Bevy 0 13 (#268)

### PR DESCRIPTION
**Pull request by jpedrick**
Compiles with [bevy from git](https://github.com/bevyengine/bevy/commit/bc98333d7c53e971b3fe0dd3122c98d99671b2a7)

Solves #268 

Refactored examples to make world explorer optional and remove some of the duplicate boilerplate app setup.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,8 @@ default = ["2d", "3d", "gpu_tests"]
 # This is a testing-only feature, which has no effect on the build.
 gpu_tests = []
 
+world_inspector = ["bevy-inspector-egui"]
```
Comment (jpedrick) [inline Cargo.toml:30]: I wanted to defer updating `bevy-inspector-egui` to Bevy main, so I made this an optional feature.

File: examples/2d.rs [hunk 2]
```diff
@@ -5,44 +5,22 @@
...
-use bevy_inspector_egui::quick::WorldInspectorPlugin;
+mod example_features;
 
 use bevy_hanabi::prelude::*;
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let mut wgpu_settings = WgpuSettings::default();
-    wgpu_settings
-        .features
-        .set(WgpuFeatures::VERTEX_WRITABLE_STORAGE, true);
```
Comment (jpedrick) [inline examples/2d.rs:23]: Most of the examples set this, so I moved it to "ExampleFeaturesPlugin". However, I wonder if there's a nicer way to wrap this for bevi_hanabi:

```rust
app.add_plugins(DefaultPlugins.set(hanabi_render_settings(RenderPlugin))
```

File: Cargo.toml (hunk 1 above)
Comment (djeedai) [reply Cargo.toml:30]: I tried in the past and that didn't work. I don't remember exactly why though.

File: Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "bevy_hanabi"
-version = "0.10.0-dev"
+version = "0.10.1-dev"
```
Comment (djeedai) [inline Cargo.toml:3]: Upgrades to new Bevy version are always breaking changes, this can't be 0.10.x

File: examples/2d.rs (hunk 2 above)
Comment (djeedai) [reply examples/2d.rs:23]: All those examples changes are unrelated to the Bevy 0.13 upgrade though, shouldn't be in the same PR.

File: examples/Makefile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+               run-spawn_on_command \
+               run-activate \
+               run-force_field \
+               run-lifetime \
+               run-instancing \
+               run-gradient \
+               run-billboard \
+               run-2d
+
+broken-in-debug-examples: run-init \
```
Comment (djeedai) [inline examples/Makefile:38]: "broken in debug" means the PR won't be merged.

File: src/spawn.rs [hunk 5]
```diff
@@ -438,6 +438,21 @@ impl EffectSpawner {
         self.active
     }
 
+    /// Get whether the spawner has more to spawn
+    ///
+    /// Spawners with infinite limit will not spawn after `spawn_remainder < 1.`
+    pub fn has_more_to_spawn(&self) -> bool {
```
Comment (djeedai) [inline src/spawn.rs:444]: Unrelated to Bevy 0.13 update.

**Review by djeedai — COMMENTED** [review]
This PR is a mixed bag of unrelated changes. I'll likely grab the 0.13 part and test it on my own. Thanks for the work though, this makes the upgrade a lot easier.

**Comment by jpedrick** [comment]
> This PR is a mixed bag of unrelated changes. I'll likely grab the 0.13 part and test it on my own. Thanks for the work though, this makes the upgrade a lot easier.

Ok sure, yeah, the example changes are largely related to the 0.13 upgrade though. I was modifying each and every example, so I refactored a bit to separate the boilerplate.

File: examples/Makefile (hunk 4 above)
Comment (jpedrick) [reply examples/Makefile:38]: Yeah, I wanted to visibly draw attention to the issue.

File: src/spawn.rs (hunk 5 above)
Comment (jpedrick) [reply src/spawn.rs:444]: This is the one that's actually unrelated. I can back it out if you'd like.

**Comment by jpedrick** [comment]
> Closing this, support for Bevy 0.13 was merged. Thanks @jpedrick for the contributions though, I used your work and that saved me quite some time! :)

Glad I could help!

**Pull request closed without merging**